Flaky test: dispatch-continuation "suppresses continuations after a raw FSM workflow reaches a terminal node" races under parallel load

## Summary

`tests/dispatch-continuation.test.ts > dispatch continuation cap > suppresses continuations after a raw FSM workflow reaches a terminal node` intermittently fails in CI under the full parallel suite.

Observed on PR #209's CI run [26291282061](https://github.com/pmatos/symphonika/actions/runs/26291282061):

- **Test (Node 24): FAILED** — `tests/dispatch-continuation.test.ts:474`
  ```
  AssertionError: expected "vi.fn()" to not be called at all, but actually been called 1 times
  ❯ tests/dispatch-continuation.test.ts:474:44
      expect(githubIssuesApi.getIssue).not.toHaveBeenCalled();
  ```
- **Test (Node 25): PASSED** — identical code, same run.

It also passes locally **3/3** when run in isolation (`npx vitest run tests/dispatch-continuation.test.ts -t "suppresses continuations after a raw FSM workflow reaches a terminal node"`, Node v26.2.0).

## Why it's flaky

The test starts a real daemon, waits for a run to reach `succeeded`, sleeps `80ms` (continuation delay is `5ms`), then asserts `getIssue` was **never** called — relying on the raw-FSM terminal suppression short-circuiting `scheduleNext` before the label refresh. Under parallel load on a contended runner, a continuation can be scheduled before the terminal suppression takes effect, so `getIssue` fires once. The pass/fail outcome is timing-dependent (per-process scheduling jitter), not logic-dependent on the code under test.

## Notes

- The test file is **not** modified by PR #209; this is pre-existing flakiness (already noted in #209's description as "known prepared-workspace flakiness"). The PR-209 CI failure was cleared by re-running the failed job.
- Repeatedly surfaces as a red check that blocks otherwise-green PRs.

## Suggested fix directions

- Make the assertion event-driven instead of sleep-based: wait for a deterministic signal that `scheduleNext` has run and been suppressed, rather than racing an `80ms` timer against a `5ms` continuation delay.
- Or inject a controllable clock / await the suppression decision directly so the test no longer depends on wall-clock ordering under load.

_Filed by `pm-autofix-pr` while converging PR #209; the failing job there was a re-run-to-green, but the underlying flake deserves its own fix._

--- a/src/lifecycle/run-controller.ts
+++ b/src/lifecycle/run-controller.ts
@@ -3014,6 +3014,14 @@ export class RunController {
     // `agent-ready`. Markdown compatibility-graph workflows keep the legacy
     // "loop on agent-ready" behavior.
     if (input.suppressContinuation === true) {
+      this.logger?.info(
+        {
+          issueNumber: input.issue.number,
+          project: input.project.name,
+          runId: input.runId
+        },
+        "symphonika workflow suppressed label-driven continuation"
+      );
       return;
     }
 

--- a/tests/dispatch-continuation.test.ts
+++ b/tests/dispatch-continuation.test.ts
@@ -446,9 +446,8 @@ describe("dispatch continuation cap", () => {
 
     // Issue stays agent-ready after the run — without the raw-FSM terminal
     // suppression, scheduleNext would refresh, see the label, and dispatch a
-    // continuation. Polling after the first tick sees the operational claim,
-    // which keeps reconciliation from making an unrelated getIssue call while
-    // still excluding the issue from fresh dispatch.
+    // continuation. Later polls include the operational claim so the issue is
+    // still excluded from a second fresh dispatch.
     const githubIssuesApi = {
       addLabelsToIssue: vi.fn().mockResolvedValue(undefined),
       getIssue: vi.fn().mockResolvedValue({
@@ -466,6 +465,8 @@ describe("dispatch continuation cap", () => {
     const prepareIssueWorkspace = vi.fn((): Promise<PreparedIssueWorkspace> =>
       Promise.resolve(prepared)
     );
+    const logger = pino({ enabled: false });
+    const logInfo = vi.spyOn(logger, "info");
 
     const daemon = await startDaemon({
       agentProviders: { codex: provider },
@@ -474,7 +475,7 @@ describe("dispatch continuation cap", () => {
       env: { GITHUB_TOKEN: "secret-token" },
       githubIssuesApi,
       lifecyclePolicy: fastContinuationPolicy,
-      logger: pino({ enabled: false }),
+      logger,
       port: 0,
       prepareIssueWorkspace
     });
@@ -484,10 +485,6 @@ describe("dispatch continuation cap", () => {
         runs.some((run) => run["state"] === "succeeded")
       );
 
-      // Wait beyond the continuation delay (5ms in fastContinuationPolicy) to
-      // confirm no continuation gets dispatched.
-      await new Promise((resolve) => setTimeout(resolve, 80));
-
       const status = (await fetch(`${daemon.url}/api/status`).then((r) =>
         r.json()
       )) as {
@@ -496,12 +493,20 @@ describe("dispatch continuation cap", () => {
       expect(status.runs).toHaveLength(1);
       expect(status.runs[0]?.["isContinuation"]).toBe(false);
       expect(status.runs[0]?.["state"]).toBe("succeeded");
-      // getIssue should not have been called — suppression short-circuits
-      // scheduleNext before the refresh.
-      expect(githubIssuesApi.getIssue).not.toHaveBeenCalled();
     } finally {
+      // stop() drains the dispatch lifecycle, so the assertion below observes
+      // scheduleNext's decision without racing a wall-clock delay.
       await daemon.stop();
     }
+
+    expect(logInfo).toHaveBeenCalledWith(
+      expect.objectContaining({
+        issueNumber: baseIssue.number,
+        project: "symphonika",
+        runId: "run-fsm-terminal"
+      }),
+      "symphonika workflow suppressed label-driven continuation"
+    );
   });
 });
 

